### PR TITLE
Reorganisation of e2e tests

### PR DESCRIPTION
**Pull request by youknowriad**
related #17745 

This PR just splits the e2e-tests package into three disctinct folders: editor, experimental and local like proposed in #17745

### CODE REVIEW

**Comment by gziolo** [comment]
> - Editor: All stable editor features that can be run from Core without the plugin as well.
> - Experimental: Tests for experimental features waiting to be marked as stable.
> - Local: Tests that are only useful in the plugin (like the demo page), to be honest, these should probably not be in the package at all.

You can move the local and experimental test(s) to `test/e2e/specs` if that helps.

In the `Editor` group you still have this folder which requires plugins to be installed. The issue with them is that they won't work with the standalone WordPress instance. It might make sense to keep them isolated in their own group somehow.

**Comment by youknowriad** [comment]
> You can move the local and experimental test(s) to test/e2e/specs if that helps.

Yeah, I think it makes sense outside of the reusable package, that said, it's "nice" to have all the e2e tests live close together, so I kept them there for now and used a separate folder.

> In the Editor group you still have this folder which requires plugins to be installed. The issue with them is that they won't work with the standalone WordPress instance. It might make sense to keep them isolated in their own group somehow.

Yes, it contains three folders at the moment (like before): plugins, blocks and various.

---

My plan here is to land this small reorganisation and try to run the "editor" tests in Core after the next npm release.

File: packages/e2e-tests/jest.config.js [hunk 1]
```diff
@@ -12,6 +12,6 @@ module.exports = {
 	testPathIgnorePatterns: [
 		'/node_modules/',
 		'/wordpress/',
-		'e2e-tests/specs/performance.test.js',
+		'e2e-tests/specs/performance/',
```
Comment (gziolo) [inline packages/e2e-tests/jest.config.js]: You probably can move:
https://github.com/WordPress/gutenberg/blob/7e9df78eaa77c94ce5228695445868ce8f2757fd/packages/e2e-tests/specs/.gitignore to the `performance` folder.

**Review by gziolo — APPROVED** [review]
Do we have any tests for features guarded with `GUTENBERG_PHASE` flag, no?

I like all the changes introduced. Let's give it a spin and test with WP core.

**Comment by aduth** [comment]
>Yes, it contains three folders at the moment (like before): plugins, blocks and various.

It's not a big concern, but "various" feels like having a folder for the sake of having a folder. If a test cannot be classified, why not just avoid classifying it (i.e. keep in the parent directory) rather than create a classification for things which cannot be classified?

**Comment by youknowriad** [comment]
> It's not a big concern, but "various" feels like having a folder for the sake of having a folder. 

That is true, My thinking was "consistency" if we have tests in the parent folder and other in subfolders, people might need to think about which folder is best instead of just adding everything in the top folder. I don't feel very strongly though.

**Pull request merged**